packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,441 @@
+}
+
+interface ComboboxProps
+  extends React.InputHTMLAttributes<
+    Omit<HTMLInputElement, 'value' | 'defaultValue'>
+  > {
+  label: ContentNode;
+  options?: ComboboxOption[];
+  value?: ComboboxValue;
+  defaultValue?: ComboboxValue;
+  requiredText?: React.ReactNode;
+  error?: React.ReactNode;
+  autocomplete?: 'none' | 'manual' | 'automatic';
+  onSelectionChange?: <T extends HTMLElement = HTMLElement>({
+    target,
+    value,
+    previousValue
+  }: {
+    target: T;
+    value: ComboboxValue;
+    previousValue: ComboboxValue;
+  }) => void;
+  onActiveChange?: (option: ListboxOption) => void;
+  renderNoResults?: (() => JSX.Element) | React.ReactElement;
+  portal?: React.RefObject<HTMLElement> | HTMLElement;
+}
+
+const defaultAutoCompleteMatches = (inputValue: string, value: string) => {
+  // istanbul ignore if
+  if (!value) {
+    return true;
+  }
+  return value.toLowerCase().includes(inputValue.toLowerCase());
+};
+
+const ComboboxNoResults = (): JSX.Element => {
+  return (
+    <div className="ComboboxListbox__empty" aria-live="polite">
+      No results found.
+    </div>
+  );
+};
+
+const Combobox = forwardRef<HTMLInputElement, ComboboxProps>(
+  (
+    {
+      id: propId,
+      className,
+      label,
+      children,
+      options = [],
+      value: propValue,
+      defaultValue,
+      requiredText = 'Required',
+      error,
+      autocomplete = 'manual',
+      onSelectionChange,
+      onActiveChange,
+      onChange,
+      onKeyDown,
+      onFocus,
+      onBlur,
+      renderNoResults,
+      portal,
+      ...props
+    },
+    ref
+  ): JSX.Element => {
+    const [value, setValue] = useState<string>(defaultValue || propValue || '');
+    const [matchingOptions, setMatchingOptions] = useState<
+      Map<HTMLElement, ComboboxOptionState>
+    >(new Map());
+    const [selectedValue, setSelectedValue] = useState<string>(value || '');
+    const [open, setOpen] = useState(false);
+    const [activeDescendant, setActiveDescendant] =
+      useState<ListboxOption | null>(null);
+    const [id] = propId ? [propId] : useId(1, 'combobox');
+    const comboboxRef = useSharedRef<HTMLDivElement>(ref);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const listboxRef = useRef<HTMLUListElement>(null);
+    const isControlled = typeof propValue !== 'undefined';
+    const isRequired = !!props.required;
+    const isAutoComplete = autocomplete !== 'none';
+    const hasError = !!error;
+
+    const comboboxOptions =
+      children ||
+      options.map((option, index) => (
+        <ComboboxOption
+          key={option.key || index}
+          id={`${id}-option-${index + 1}`}
+          description={option.description}
+        >
+          {option.label}
+        </ComboboxOption>
+      ));
+
+    useEffect(() => {
+      if (!isAutoComplete) {
+        return;
+      }
+
+      if (!open && selectedValue && value !== selectedValue) {
+        setValue(selectedValue);
+      }
```

Comment:
Playing around with it and looking at this, if I understand this correctly, the input's value is set to the value of the combobox option value. However, if the combobox option `value` is something like "option-1" and the option's `children` is "Pizza", it seems like we should display "Pizza"--not "option-1"

Reply:
My original thinking is that `value` is the value you want as text in the resulting Combobox input and is the value that gets submitted to the server, while `label` is the text that gets displayed in the listbox. Text matching happens based on the label, not the value, and from looking at other comboboxes this is generally how they work with a few exceptions.

Although you bring up a good point that for non-text values that are intended to be internal we might want something different here. For an example, [Adobe's combobox has a `formValue` property](https://react-spectrum.adobe.com/react-spectrum/ComboBox.html#:~:text=space%20is%20limited.-,formValue,-%27text%27%20%7C%C2%A0) to specify the value that gets submitted as part of a form. We might could do something similar here, but I wonder if it's critical enough to be a part of this PR or we can tackle that separately.

I would really only consider this blocking if the change would be breaking from the existing pattern. Let me mull on this for a bit.

Reply:
Proposal/RFC: https://github.com/dequelabs/cauldron/issues/1184